.github/workflows/autolabel-pullrequests.yml

```diff
@@ -0,0 +1,117 @@
+---
+name: Auto Label the Pull Requests with the component names
+on:  # yamllint disable-line rule:truthy
+  # Use pull_request_target to gain write permissions.
+  # Ref: https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/
+  pull_request_target:
+    types: [opened, reopened, synchronize]
+    paths:
+      - 'ci-scripts/**'
+      - '.circleci/**'
+      - 'circleci/**'
+      - '.github/**'
+      - 'cn/**'
+      - 'cwf/**'
+      - 'docs/**'
+      - 'example/**'
+      - 'experimental/**'
+      - 'fbinternal/**'
+      - 'feg/**'
+      - 'lte/**'
+      - 'nms/**'
+      - 'openwrt/**'
+      - 'orc8r/**'
+      - 'protos/**'
+      - 'scripts/**'
+      - 'secrets/**'
+      - 'show-tech/**'
+      - 'third_party/**'
+      - 'wifi/**'
+      - 'xwf/**'
+
+jobs:
+  comment:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v3
+        with:
+          script: |
+
+            let lbl;
+            let lbls = new Set(); // Set of label strings
+
+            const pulledFiles = await github.pulls.listFiles({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number,
+            });
+
+            for (const f of pulledFiles['data']) {
+             switch (true) {
```

Comment:
1. It would be preferable to pull this logic into a separate yaml config file, then this function can be "dumb" and just read in and operate based on that yaml config file
2. Unless I'm mistaken, looks like this only applies 1 label per PR (?). We actually want all the relevant component labels for every component the pr changes

cc @ranj8work @quentinDERORY